linker/shared/src/main/scala/org/scalajs/linker/analyzer/Analyzer.scala

```diff
@@ -640,10 +709,14 @@ private final class Analyzer(config: CommonPhaseConfig,
         right: ir.Types.TypeRef): Boolean = {
       import ir.Types._
 
+      def getClassInfo(name: String): Option[ClassInfo] =
+        // TODO: This is suspicious: It shouldn't be necessary.
```

Comment:
Any idea why/when it's necessary? What should be the expected implementation?

Reply:
It is mainly the original code that is suspicious: It assumes the class is already loaded but allows and ignores non-existent classes. That doesn't make any sense to me.

In fact, it seems that this also works around the limitation that note all type refs are linked. I'll have a look.

Reply:
So, I had a closer look at this and it seems that this (and the previous code) simply *ignores* if the result type of a potential reflective call target is not loaded. I suspect this could lead to non-deterministic behavior; I will try force load class infos properly here.